docs/contributing/testing/Leak-tracking.md

```diff
@@ -0,0 +1,53 @@
+# Leak tracking in Flutter framework
+
+Flutter Framework uses [leak_tracker](https://github.com/dart-lang/leak_tracker/blob/main/doc/leak_tracking/OVERVIEW.md) to auto-detect not disposed objects.
+
+This page contains Flutter Framework related information.
+
+See leak_tracker documentation for
+[general leak troubleshooting](https://github.com/dart-lang/leak_tracker/blob/main/doc/leak_tracking/TROUBLESHOOT.md).
+
+## When it is ok to opt-out from leak tracking?
+
+In general, leak tracking should be enabled, to verify that all
+disposables are disposed.
+All exceptions should be clearly explained in the comments.
+
+It is ok to opt out a test from leak tracking in these cases:
+
+1. Hacky tests
+
+    A test leaks because it uses APIs in non-recommended way
+    and there is no simple way to either test what is needed
+    in a recommended way or to dispose the objects.
+
+2. Thrown exceptions
+
+    Some leaks happen because exceptions are thrown and
+    the code did not finalize properly.
+
+    While some exceptions should be finalized properly
+    and should not result in leaking objects,
+    the project to make sure disposables are disposed
+    in exceptional code paths was not prioritized yet.
```

Comment:
Is there an issue tracking this that folks can refer to for updates?

Reply:
Created: https://github.com/flutter/flutter/issues/157470